modules/login/php/ajaxlogin.class.inc

```diff
@@ -0,0 +1,383 @@
+            $response = $this->expiredPassword($values);
+        } else {
+            $response['status'] = 'error';
+        }
+        return $response;
+    }
+
+    /**
+     * Initialize setup.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function initializeSetup(array $values) : array
+    {
+        // Login Setup.
+        $loginData = array();
+        $login     = $_SESSION['State']->getProperty('login');
+
+        // Previous login failed.
+        if (!empty($login->_lastError)) {
+            $loginData['error'] = $login->_lastError;
+        }
+        $config      =& \NDB_Config::singleton();
+        $study_links = $config->getExternalLinks('Studylinks');
+        foreach ($study_links as $label => $url) {
+            $windowName           = md5($url);
+            $loginData['links'][] = array(
+                                     'url'        => $url,
+                                     'label'      => $label,
+                                     'windowName' => $windowName,
+                                    );
+        }
+        $loginData['description'] = html_entity_decode(
+            $config->getSetting('StudyDescription')
+        );
+        $loginData['title']       = $config->getSetting('title');
+        $loginData['logo']        = $config->getSetting('studylogo');
+        $values['login']          = $loginData;
+
+        // Request Account Setup.
+        $requestAccountData = array();
+        $reCAPTCHAPrivate   = $config->getSetting('reCAPTCHAPrivate');
+        $reCAPTCHAPublic    = $config->getSetting('reCAPTCHAPublic');
+        // reCAPTCHA if both private and public keys are set
+        if ($reCAPTCHAPrivate && $reCAPTCHAPublic) {
+            $requestAccountData['captcha'] = $reCAPTCHAPublic; // captcha key.
+        }
+        $requestAccountData['site'] = \Utility::getSiteList();
+        $values['requestAccount']   = $requestAccountData;
+
+        return $values;
+    }
+
+    /**
+     * Reset password.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function resetPassword(array $values) : array
+    {
+        // Always claim success to avoid providing user info to attackers.
+        $values['success'] = 'Password reset. '
+            . 'You should receive an email within a few minutes.';
+
+        // Create the user object.
+        $username = $values['username'];
+        $user     =& \User::singleton($username);
+
+        // Check if valid user.
+        $email = $user->getData('Email');
+        if (empty($email)) {
+            $values['error'] = 'Not a valid user';
+            return $values;
+        }
+
+        try {
+            // Check if valid email.
+            if ($user->isEmailValid()) {
+                $config = \NDB_Config::singleton();
+                // Generate a new password.
+                $password = \User::newPassword();
```

Comment:
This function doesn't exist anymore. You'll need to take a deeper look into #4411 and rework the password/login logic in this PR to use the logic from the new Password class.